packages/app/src/cli/services/build/extension.ts

```diff
@@ -106,6 +108,25 @@ export async function buildUIExtension(extension: ExtensionInstance, options: Ex
       stdout: options.stdout,
       sourceMaps: extension.isSourceMapGeneratingExtension,
     })
+    if (assets) {
+      await Promise.all(
+        assets.map(async (asset) => {
+          await bundleExtension({
+            minify: true,
+            outputPath: joinPath(dirname(extension.outputPath), `${extension.configuration.handle}-conditions.js`),
```

Comment:
In here you are not using the output path defined in the asset, this should be `asset.filepath` right?
But then you'll have the same problem of the double nested `dist/dist` folder.